attributes/default.rb

```diff
@@ -18,3 +18,7 @@
 #
 
 default['build-essential']['compile_time'] = false
+
+# Solaris 10 omnibus-build-essential package information
+default['build-essential']['solaris10_package_url'] = 'https://chef-releng.s3.amazonaws.com/omnibus/build-essential'
+default['build-essential']['solaris10_package_version'] = '0.0.4-1'
```

Comment:
I think you can simplify everything with something like this:

```ruby
case node['platform_family']
when 'solaris2'
  default['build-essential']['package_url']      = "https://chef-releng.s3.amazonaws.com/omnibus/build-essential-0.0.4-1-#{node['kernel']['machine']}.solaris"
  default['build-essential']['package_checksum'] = 'SOME_SHA256'
end
```

You can then get the package name in the recipe with:

```ruby
omnibus_package_name = File.basename(node['build-essential']['package_url'])
```